pkg/ebpf/tracee.go

```diff
@@ -1252,11 +1290,17 @@ func (t *Tracee) writeProfilerStats(wr io.Writer) error {
 }
 
 func (t *Tracee) getProcessCtx(hostTid uint32) (procinfo.ProcessCtx, error) {
+
+	//FIXME: is this access going to affect performance?
```

Comment:
This function is invoked as a hot path only in the network events which are getting replaced.
So I think this path is getting deleted anyway in the near future (@yanivagman @rafaeldtinoco  correct me if i'm wrong).

Reply:
@yanivagman I see the main bpf module as a thing we'll eventually deprecate as all logic will eventually be separated into different modules.

Reply:
Have what I think is a better approach started here: https://github.com/grantseltzer/tracee/tree/modular-tracee-bpf-register

i'll open that in a PR tomorrow.

Reply:
@yanivagman What do you think of the current state of this PR? The next step after this is moving maps from the main module to a separate map one.